src/request/change-password/ChangePassword.js

```diff
@@ -179,28 +136,63 @@ class ChangePassword {
         TopLevelApi.setLoading(false);
     }
 
+    async _prepare() {
+        let colorClass = '';
+        if (this.key.secret instanceof Nimiq.Entropy) {
+            const color = Iqons.getBackgroundColorIndex(
+                new Nimiq.Address(
+                    // use color of first address as loginFile color
+                    this.key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+                ).toUserFriendlyAddress(),
+            );
+            const colorString = LoginFile.CONFIG[color].name;
+            colorClass = `nq-${colorString}-bg`;
+        }
+        this._loginFileIcon.lock(colorClass);
+        this._progressIndicator.setStep(3);
+    }
+
     /**
-     * @param {string} phrase
+     * @param {string} newPassword
      */
-    async _finish(phrase) {
-        TopLevelApi.setLoading(true);
-        if (!this._key) {
-            this._reject(new Errors.KeyguardError('Bypassed Password'));
-            return;
-        }
+    async _finish(newPassword) {
```

Comment:
This function does not actually _finish_ the request for entropies. Those continue to the LoginFile download page. Thus `_finish` might be misleading.